### PR TITLE
New Flex cheat sheet

### PR DESCRIPTION
**Pull request by hell-sing**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [x] New Instant Answer
  - [x] Cheat Sheets: **`New {flex} Cheat Sheet`**
  - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
  - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
  - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

This PR adds Flex Cheat Sheet.
###### Which issues (if any) does this fix?

None
###### People to notify (@mention interested parties)

@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/flex_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@hell-sing Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by hell-sing** [comment]
@gautamkrishnar The travis seems to pass the build but it is still showing travis failed. Please let me know about the issue. TIA.

**Comment by gautamkrishnar** [comment]
@hell-sing its not because of your cheatsheet. Its because of another IA.

**Comment by hell-sing** [comment]
@gautamkrishnar How to correct thing. Please help.TIA.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=flex  cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/flex_cheat_sheet_mobile.png?VBmB2zCUop)

**Comment by gautamkrishnar** [comment]
@hell-sing sorry, I cant find the exact cause of it. Its some dependency problem. Its an internal problem. Lets wait for someone from the community to fix it. Just ignore the travis test for now...

File: share/goodie/cheat_sheets/json/flex.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            {
+                "key": "ws",
+                "val": "\\{delim\\}+"
+            }
+        ],
+        "Interfacing with YACC/Bison": [
+            {
+                "key": "#include<y.tab.h>",
+                "val": "Contain all the definitions of %token appearing in yacc file and must be included in lex.l"
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/flex.json]: remove the `%` symbol

File: share/goodie/cheat_sheets/json/flex.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            {
+                "key": "auxiliary functions",
+                "val": "This section includes additional functions"
+            }
+        ],
+        "Regular Definitions": [
+            {
+                "key": "digit",
+                "val": "\\[0-9\\]"
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/flex.json]: you are escaping the square brackets incorrectly. Please change this to `\[0-9\]`

File: share/goodie/cheat_sheets/json/flex.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            {
+                "key": "letter",
+                "val": "\\[A-za-z\\]"
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/flex.json]: `\[A-za-z\]`

File: share/goodie/cheat_sheets/json/flex.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            },
+            {
+                "key": "number",
+                "val": "\\{digit\\}+(\\.\\{digit\\}+)?(E\\[+-\\]?\\{digit\\}+)?"
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/flex.json]: `Please correct these... You only need to use `\[` for displaying `[` and `\{` for displaying `\{`.
You can use `\\` if you need to display `\`.
Use `\"` to escape quotes...

File: share/goodie/cheat_sheets/json/flex.json [hunk 5]
```diff
@@ -0,0 +1,84 @@
+{
+  "id": "flex_cheat_sheet",
+  "name": "flex",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/flex.json]: make this "Flex"

File: share/goodie/cheat_sheets/json/flex.json [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  "description": "flex-fast lexical analyzer generator, a tool for generating programs that perform pattern-matching on text",
+  "metadata": {
+    "sourceName": "Dinosaur.compilertools.net",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/flex.json]: "compilertools.net" Would be enough here

File: share/goodie/cheat_sheets/json/flex.json (hunk 3 above)
Comment (hell-sing) [inline share/goodie/cheat_sheets/json/flex.json]: @gautamkrishnar I need output like [A-za-z] what will be the code for this in "val": section.

File: share/goodie/cheat_sheets/json/flex.json (hunk 6 above)
Comment (hell-sing) [inline share/goodie/cheat_sheets/json/flex.json]: @jophab thank you for your suggestion.I will change that. 👍

File: share/goodie/cheat_sheets/json/flex.json (hunk 1 above)
Comment (hell-sing) [inline share/goodie/cheat_sheets/json/flex.json]: @gautamkrishnar There is no response from anyone in the Name Days: Failing Travis CI tests #3766 . Can you make things work fast. I really want to see my cheat sheet in action. Thank you for your efforts.

File: share/goodie/cheat_sheets/json/flex.json (hunk 3 above)
Comment (hell-sing) [inline share/goodie/cheat_sheets/json/flex.json]: @gautamkrishnar  please help with this( above comment ).Thank You.

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/flex.json]: @hell-sing  just use `\[A-za-z\]' as val

File: share/goodie/cheat_sheets/json/flex.json (hunk 5 above)
Comment (hell-sing) [inline share/goodie/cheat_sheets/json/flex.json]: @jophab Doen. Thank you

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=flex  cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/flex_cheat_sheet_mobile.png?6woxh9y0TI)

**Comment by hell-sing** [comment]
@gautamkrishnar Done some changes.Please Check. The statement in val and all the brackets in val is just for printing purpose nothing more. I want to print them just as it is. Please tell me if you want anything to be changed. Thank you for you time and effort.

**Comment by gautamkrishnar** [comment]
@hell-sing i will try to edit the file for you 👍

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=flex  cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/flex_cheat_sheet_mobile.png?EGEPOwQI40)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=flex  cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/flex_cheat_sheet_mobile.png?XM7ULmMRgp)

**Comment by gautamkrishnar** [comment]
@moollaza i think the commit is not yet deployed. 
Is there any change in the escaping of the symbols? 
cc: @GuiltyDolphin

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=flex  cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/flex_cheat_sheet_mobile.png?6hp4tTIpWa)

**Comment by gautamkrishnar** [comment]
@hell-sing https://beta.duckduckgo.com/?q=flex+cheat+sheet&ia=cheatsheet&iax=1
is this okay now?

**Comment by hell-sing** [comment]
@gautamkrishnar yes everything is okay. Thank you for your time and effort. 👍

**Comment by daxtheduck** [comment]
@hell-sing this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by gautamkrishnar** [comment]
Congrats! @hell-sing 🍿

**Pull request merged**